src/uct/base/uct_iface.h

```diff
@@ -412,23 +412,37 @@ struct uct_iface_config {
  */
 typedef struct uct_iface_mpool_config {
     unsigned          max_bufs;  /* Upper limit to number of buffers */
-    unsigned          bufs_grow; /* How many buffers (approx.) are allocated every time */
+    unsigned          bufs_grow; /* How many buffers (approx.) are allocated 1st time */
+    unsigned          max_chunk_size;
+    double            grow_factor; /* Increase each new allocated chunk by this factor */
 } uct_iface_mpool_config_t;
 
 
 /**
  * Define configuration fields for memory pool parameters.
  */
-#define UCT_IFACE_MPOOL_CONFIG_FIELDS(_prefix, _dfl_max, _dfl_grow, _mp_name, _offset, _desc) \
+#define UCT_IFACE_MPOOL_CONFIG_FIELDS(_prefix, _dfl_max, _dfl_grow, \
+                                      _dfl_max_chunk_size, _dfl_grow_factor, \
+                                      _mp_name, _offset, _desc) \
     {_prefix "MAX_BUFS", UCS_PP_QUOTE(_dfl_max), \
      "Maximal number of " _mp_name " buffers for the interface. -1 is infinite." \
      _desc, \
      (_offset) + ucs_offsetof(uct_iface_mpool_config_t, max_bufs), UCS_CONFIG_TYPE_INT}, \
     \
     {_prefix "BUFS_GROW", UCS_PP_QUOTE(_dfl_grow), \
-     "How much buffers are added every time the " _mp_name " memory pool grows.\n" \
+     "The initial chunk size of " _mp_name " memory pool.\n" \
      "0 means the value is chosen by the transport.", \
-     (_offset) + ucs_offsetof(uct_iface_mpool_config_t, bufs_grow), UCS_CONFIG_TYPE_UINT}
+     (_offset) + ucs_offsetof(uct_iface_mpool_config_t, bufs_grow), UCS_CONFIG_TYPE_UINT}, \
+    \
+    {_prefix "MAX_CHUNK_SIZE", UCS_PP_QUOTE(_dfl_max_chunk_size), \
+     "Maximal size for new chunks of " _mp_name ". memory pool.\n" \
+     "Only take effect on grow_factor equal to 1.", \
+     (_offset) + ucs_offsetof(uct_iface_mpool_config_t, max_chunk_size), UCS_CONFIG_TYPE_UINT}, \
+    \
+    {_prefix "GROW_FACTOR", UCS_PP_QUOTE(_dfl_grow_factor), \
+     "Growth factor for new chunks in " _mp_name ". Each time a new chunk is allocated,\n" \
+     "its size is the multiple of the previous chunk, by this number.",\
```

Comment:
"its size is the multiple of the previous chunk size by this number.",

Reply:
fixed